build-support/bin/native/bootstrap_code.sh

```diff
@@ -0,0 +1,106 @@
+REPO_ROOT=$(cd $(dirname "${BASH_SOURCE[0]}") && cd ../../.. && pwd -P)
+
+# Defines:
+# + CACHE_ROOT: The pants cache directory, ie: ~/.cache/pants.
+# Exposes:
+# + die: Exit in a failure state and optionally log an error message to the console.
+# + fingerprint_data: Fingerprints the data on stdin.
+source ${REPO_ROOT}/build-support/common.sh
+
+source ${REPO_ROOT}/build-support/bin/native/bootstrap_rust.sh
+
+readonly KERNEL=$(uname -s | tr '[:upper:]' '[:lower:]')
+case "${KERNEL}" in
+  linux)
+    readonly LIB_EXTENSION=so
+    ;;
+  darwin)
+    readonly LIB_EXTENSION=dylib
+    ;;
+  *)
+    die "Unknown kernel ${KERNEL}, cannot bootstrap pants native code!"
+    ;;
+esac
+
+readonly NATIVE_ROOT="${REPO_ROOT}/src/rust/engine"
+readonly NATIVE_ENGINE_BINARY="native_engine.so"
+readonly NATIVE_ENGINE_RESOURCE="${REPO_ROOT}/src/python/pants/engine/${NATIVE_ENGINE_BINARY}"
+
+# N.B. Set $MODE to "debug" to generate a binary with debugging symbols.
```

Comment:
Our release mode binaries actually still have debug symbols in them (see a comment in `src/rust/engine/Cargo.toml`'s `profile.release` section). `--release` just controls optimisation (and so compile/link times)

Reply:
I just carried this comment along in the refactor move, but I'll fix.